packages/opentelemetry-exporter-collector/src/platform/node/types.ts

```diff
@@ -17,17 +17,26 @@
 import * as grpc from 'grpc';
 import { ReadableSpan } from '@opentelemetry/tracing';
 import { CollectorExporterError } from '../../types';
-
+import { MetricRecord } from '@opentelemetry/metrics';
 /**
  * Queue item to be used to save temporary spans in case the GRPC service
  * hasn't been fully initialised yet
  */
-export interface GRPCQueueItem {
+export interface GRPCSpanQueueItem {
   spans: ReadableSpan[];
   onSuccess: () => void;
   onError: (error: CollectorExporterError) => void;
 }
 
+/**
+ * Queue item to be used to save temporary metrics
+ */
+export interface GRPCMetricQueueItem {
```

Comment:
Could these two interfaces both extend a base QueueItem interface that contains the onSucces, onError callbacks and just add the spans and metrics fields accordingly?